comp/core/healthprobe/healthprobeimpl/healthprobe_test.go

```diff
@@ -0,0 +1,113 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package healthprobeimpl implements the healthprobe component interface
+package healthprobeimpl
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/comp/core/log/logimpl"
+	"github.com/DataDog/datadog-agent/pkg/status/health"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"go.uber.org/fx"
+	"go.uber.org/fx/fxtest"
+	"gotest.tools/assert"
+)
+
+func TestServer(t *testing.T) {
+	lc := fxtest.NewLifecycle(t)
+
+	port := "7869"
+	overrides := map[string]any{
+		"health_port": port,
+	}
+
+	_, err := newHealthProbe(lc,
+		fxutil.Test[dependencies](
+			t,
+			logimpl.MockModule(),
+			config.MockModule(),
+			fx.Replace(config.MockParams{Overrides: overrides}),
+		))
+
+	assert.NilError(t, err)
+
+	ctx := context.Background()
+	assert.NilError(t, lc.Start(ctx))
+
+	assert.NilError(t, lc.Stop(ctx))
+}
+
+func TestLiveHandler(t *testing.T) {
+	configComponent := fxutil.Test[config.Component](t, config.MockModule())
+	logComponent := fxutil.Test[log.Component](t, logimpl.MockModule())
```

Comment:
A pattern that I find useful is to have a `struct deps` that contains all the components needed (See [this code](https://github.com/DataDog/datadog-agent/blob/8e00982132c765e216f5032e83a814c471d399dd/pkg/metadata/scheduler_test.go#L223-L230)). It has the nice property of having a single call to `fxutil.Test` which avoid having several instances of the same component. Adding another component is also very easy.

Reply:
Sounds good. Thank you so much for the suggestion